### PR TITLE
Add bark tts

### PR DESCRIPTION
**Pull request by ravi03071991**
Hi,

I have added the barkTTS module to TTS.

Thank you.

### CODE REVIEW

File: gpt_index/tts/base.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
+"""Text to speech module."""
+
+from typing import Optional
+
+
+class barkTTS:
```
Comment (jerryjliu) [inline gpt_index/tts/base.py:6]: Can we capitalize the class name? BarkTTS

File: gpt_index/tts/base.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+            raise ImportError(import_err_msg)
+
+        self.generate_audio = bark.generate_audio
+        self.save_as_prompt = bark.save_as_prompt
+        self.text_temp = text_temp
+        self.waveform_temp = waveform_temp
+
+    def generate_bark_audio(
+        self, text: Optional[str], lang_speaker_voice: Optional[str]
+    ) -> None:
```
Comment (jerryjliu) [inline gpt_index/tts/base.py:30]: would be good to abstract out a BaseTTS class too that takes in text as a single argument. lang_speaker_voice could probably be moved to the `__init__`

File: gpt_index/tts/base.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        """
+        Args:
+            text: text to be turned into audio.
+            lang_speaker_voice: language speaker voice for audio cloning.
+        """
+
+        import tempfile
+        import os
+        import numpy as np
```
Comment (jerryjliu) [inline gpt_index/tts/base.py:39]: why are these lazy imports? they should all be included in the package by default

File: gpt_index/tts/base.py (hunk 2 above)
Comment (jerryjliu) [reply gpt_index/tts/base.py:30]: also lang_speaker_voice could be an optional kwarg

File: gpt_index/tts/base.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+        """
+
+        import_err_msg = "`bark` package not found, \
+            please run `pip install git+https://github.com/suno-ai/bark.git`"
+        try:
+            import bark
+        except ImportError:
+            raise ImportError(import_err_msg)
+
+        self.generate_audio = bark.generate_audio
```
Comment (jerryjliu) [inline gpt_index/tts/base.py:23]: i'd call this generate_audio_fn

**Review by jerryjliu — COMMENTED** [review]
Generally looks good, some comments.

Also, why does the example notebook have 3 repeated `.ipynb` extensions?

File: gpt_index/tts/base.py (hunk 1 above)
Comment (ravi03071991) [reply gpt_index/tts/base.py:6]: Sure.

File: gpt_index/tts/base.py [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+        super().__init__()
+
+        import_err_msg = "`bark` package not found, \
+            please run `pip install git+https://github.com/suno-ai/bark.git`"
+        try:
+            import bark
+        except ImportError:
+            raise ImportError(import_err_msg)
+
+        self.generate_audio_fn = bark.generate_audio
```
Comment (Disiok) [inline gpt_index/tts/base.py:39]: nit: could you use e.g. `self._generate_audio_fn` to mark these attributes as private?

Comment (Disiok) [reply gpt_index/tts/base.py:39]: Unless you have a strong reason, it's better to reveal as little as possible about the underlying implementation of an object.

File: gpt_index/tts/base.py [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+        self.save_as_prompt = bark.save_as_prompt
+        self.text_temp = text_temp
+        self.waveform_temp = waveform_temp
+        self.lang_speaker_voice = lang_speaker_voice
+
+    def generate_bark_audio(self, text: str) -> None:
```
Comment (Disiok) [inline gpt_index/tts/base.py:45]: Could we revise this to `def generate_audio` and have this as part of the interface defined by `BaseTTS`?

This way you can call different implementations of `BaseTTS`

Comment (Disiok) [inline gpt_index/tts/base.py:45]: output type seems wrong?

File: gpt_index/tts/base.py [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+        words = text.split()
+        chunks = [words[i : i + 10] for i in range(0, len(words), 10)]
+        chunks = [" ".join(chunk) for chunk in chunks]  # type: ignore
+
+        full_generation = None
+        history_prompt = self.lang_speaker_voice
+        audio_chunks = []
+
+        for chunk in chunks:
+            with tempfile.TemporaryDirectory() as d:
```
Comment (Disiok) [inline gpt_index/tts/base.py:61]: Curious: do we have to write to disk to call the bark API?

File: gpt_index/tts/base.py [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+
+    def generate_bark_audio(self, text: str) -> None:
+        """
+        Args:
+            text: text to be turned into audio.
+        """
+
+        # text to be chunked into 10 words each to avoid hallicunation
+        words = text.split()
+        chunks = [words[i : i + 10] for i in range(0, len(words), 10)]
```
Comment (Disiok) [inline gpt_index/tts/base.py:53]: nit: can you use a constant var e.g. `DEFAULT_CHUNK_SIZE` or make it configurable?

File: gpt_index/tts/base.py (hunk 5 above)
Comment (Disiok) [reply gpt_index/tts/base.py:39]: Also, why do we need to save bark.generate_audio as an instance variable?

**Review by Disiok — CHANGES_REQUESTED** [review]
Sweet, thanks for the contribution!

File: gpt_index/tts/base.py (hunk 5 above)
Comment (ravi03071991) [reply gpt_index/tts/base.py:39]: if not saved as an instance variable I need to `import bark` in the `generate_bark_audio` function as well. So having it as an instance variable makes sense here?

File: gpt_index/tts/base.py (hunk 7 above)
Comment (ravi03071991) [reply gpt_index/tts/base.py:61]: Yes. bark API expects `history_prompt` to be saved in a `.npz` format file.

File: gpt_index/tts/base.py (hunk 6 above)
Comment (ravi03071991) [reply gpt_index/tts/base.py:45]: sure.

Comment (ravi03071991) [reply gpt_index/tts/base.py:45]: output type is `numpy array` rather than converting it to `List` type, hence kept it as None?

**Review by jerryjliu — APPROVED** [review]

**Pull request merged**